### PR TITLE
Fix wayland cpu usage

### PR DESCRIPTION
**Pull request by jwijenbergh**
This patch reverts an optimization that was removed, so in english: it adds back an optimization :^)

Previously this optimization had issues with fullscreen/max window toggle but this patch fixes that by checking the toplevel width/height.

This optimization is needed in the wlr 0.17 codebase because we `place()` on each commit. Which is what other compositors also do and they also check for geometry changes

Fixes: #4865 #4878

### CODE REVIEW

**Comment by jwijenbergh** [comment]
@elParaguayo can you daily drive this for a bit?

**Comment by elParaguayo** [comment]
Sure. Happy to give it a go.

**Review by tych0 — APPROVED** [review]

File: libqtile/backend/wayland/xdgwindow.py [hunk 1]
```diff
@@ -287,6 +284,17 @@ def place(
...
         )
+        state = self.surface.toplevel._ptr.current
+        place_changed = any(
+            [
+                self.x != x,
+                self.y != y,
+                self._width != width,
+                self._height != height,
+                state.width != width,
+                state.height != height,
```
Comment (elParaguayo) [inline libqtile/backend/wayland/xdgwindow.py:295]: Why do we need to check `width/height` and `_width/_height`?

Comment (jwijenbergh) [reply libqtile/backend/wayland/xdgwindow.py:294]: the state's width and height is the width and height what the window reports using wlroots APIs, whilst the self._width and self._height is the width and height of the window. If any of those are mismatched we need to resize the window. the check against `self._width` and `self._height` is potentially unneeded but maybe there is some edge case to consider

**Comment by elParaguayo** [comment]
This has been running fine for me.

File: libqtile/backend/wayland/xdgwindow.py (hunk 1 above)
Comment (elParaguayo) [reply libqtile/backend/wayland/xdgwindow.py:294]: OK. It's just not clear to me how they can be different. Doesn't `self.width` just retrieve `self._width`?

Comment (jwijenbergh) [reply libqtile/backend/wayland/xdgwindow.py:294]: It says `state.width` not `self.width`

Comment (elParaguayo) [reply libqtile/backend/wayland/xdgwindow.py:294]: Ha ha. Wow. I obviously need to get my eyes tested!

Comment (jwijenbergh) [reply libqtile/backend/wayland/xdgwindow.py:294]: Haha it's fine, thanks for the review

**Pull request merged**